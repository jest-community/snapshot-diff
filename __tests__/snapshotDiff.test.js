// @flow

const React = require('react');
const snapshotDiff = require('../src/index');

const a = `
    some
    some
    some
    some
    some
    some
    some
    not
    very
    long
    script
`;
const b = `
    some
    some
    some
    some
    some
    some
    some
    not
    so
    very
    long
    script
`;

class Component extends React.Component {
  render() {
    return (
      <div>
        <span />
        <span />
        <span />
        <span>
          {this.props.test}
        </span>
        <span>
          {this.props.test}
        </span>
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span>
          {this.props.test}
        </span>
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
      </div>
    );
  }
}

test('collapses diffs and strips ansi by default', () => {
  expect(snapshotDiff(a, b)).toMatchSnapshot();
});

test('can expand diff', () => {
  expect(snapshotDiff(a, b, { expand: true })).toMatchSnapshot();
});

test('can colorize diff', () => {
  expect(snapshotDiff(a, b, { colors: true })).toMatchSnapshot();
});

test('can use contextLines on diff', () => {
  expect(snapshotDiff(a, b, { contextLines: 0 })).toMatchSnapshot();
});

test('diffs short strings', () => {
  const x = `
  abcx
  `;
  const y = `
  abcy
  `;
  expect(snapshotDiff(x, y)).toMatchSnapshot();
});

test('detects React components', () => {
  expect(
    snapshotDiff(<Component test="say" />, <Component test="my name" />)
  ).toMatchSnapshot();
});

test('can use contextLines with React components', () => {
  expect(
    snapshotDiff(<Component test="say" />, <Component test="my name" />, {
      contextLines: 0,
    })
  ).toMatchSnapshot();
});
