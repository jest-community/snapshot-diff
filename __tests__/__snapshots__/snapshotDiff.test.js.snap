// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can colorize diff 1`] = `
"Snapshot Diff:
[32m- First value[39m
[31m+ Second value[39m

[33m@@ -5,8 +5,9 @@[39m
[33m[39m[2m     some[22m
[2m     some[22m
[2m     some[22m
[2m     some[22m
[2m     not[22m
[31m+    so[39m
[2m     very[22m
[2m     long[22m
[2m     script[22m"
`;

exports[`can expand diff 1`] = `
"Snapshot Diff:
- First value
+ Second value

  
      some
      some
      some
      some
      some
      some
      some
      not
+     so
      very
      long
      script"
`;

exports[`can use contextLines on diff 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -10,0 +10,1 @@
+    so"
`;

exports[`can use contextLines with React components 1`] = `
"Snapshot Diff:
- <Component test=\\"say\\" />
+ <Component test=\\"my name\\" />

@@ -6,1 +6,1 @@
-    say
+    my name
@@ -9,1 +9,1 @@
-    say
+    my name
@@ -32,1 +32,1 @@
-    say
+    my name"
`;

exports[`collapses diffs and strips ansi by default 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -5,8 +5,9 @@
     some
     some
     some
     some
     not
+    so
     very
     long
     script"
`;

exports[`detects React components 1`] = `
"Snapshot Diff:
- <Component test=\\"say\\" />
+ <Component test=\\"my name\\" />

@@ -1,14 +1,14 @@
 <div>
   <span />
   <span />
   <span />
   <span>
-    say
+    my name
   </span>
   <span>
-    say
+    my name
   </span>
   <span />
   <span />
   <span />
   <span />
@@ -27,11 +27,11 @@
   <span />
   <span />
   <span />
   <span />
   <span>
-    say
+    my name
   </span>
   <span />
   <span />
   <span />
   <span />"
`;

exports[`diffs short strings 1`] = `
"Snapshot Diff:
- First value
+ Second value

 
-  abcx
+  abcy
   "
`;
